communication/vehicleconnections/vehicleconnection.h

```diff
@@ -35,6 +35,7 @@ class VehicleConnection : public QObject
     virtual std::string setFloatParameterOnVehicle(std::string, float value) = 0;
     virtual std::string setCustomParameterOnVehicle(std::string name, std::string value) = 0;
     virtual std::vector<std::variant<std::vector<std::pair<std::string, int32_t>>, std::vector<std::pair<std::string, float>>, std::vector<std::pair<std::string, std::string>>>> getAllParametersFromVehicle() = 0;
+    virtual bool requestRebootOrShutdownOfSystemComponents(int param = 0, int value = 0) = 0;
```

Comment:
param and value are MAVSDK-specific and not understandable without digging in the code. I think param should be an enum (autopilot, onboard computer, component) and value either bool (shutdown/reboot) or also enum. I understand that the MAVLINK lowlevel format allows setting the component ID in param, but the component id could either be a separate parameter (we do not have the same constraints here for parameters as MAVLINK messages have) or an additional entry in the param enum which we can add once we need it (e.g., camera, canopen, ...).

bool MavsdkVehicleConnection::requestRebootOrShutdownOfSystemComponents(int param, int value) should follow the same interface then.

Reply:
Should be much clearer now